File: Vestigio/code.js
var count = window.prompt("Enter the number of use cases: ");

for(let i = 0; i < count; i++){
    let useCase, n = getUseCase();
    let trace = trace(useCase, n);
    let validColum = validColum(useCase, n);
    let validRow = validRow(useCase, n);
    x
}

function getUseCase() {
    let useCase = [];
    let n = window.prompt("Enter the size of the matriz: ");
    for(let i=0; i<n; i++){
        let row = [];
        for(let j=0; i<n ; j++){
            row.push(window.prompt(`Enter the number in position ${i},${j}`));
        }
        useCase.push(row);
    }
    return useCase, n;
}

function trace(useCase, size){
    let sum = 0;
    for(let i=0; i<size; i++){
        sum += useCase[i][i];
    }
    return sum;
}

function validColum(useCase, size){
    let acum = 0;
    for(let i=0; i<size; i++){
        acum += searchRepeated(useCase[i], size);
    }
    return acum;
}

function validRow(useCase, size){
    let acum = 0;
    let rowPosition = 0;
    while(rowPosition < size){
        let row = [];
        for(let i=0; i<size; i++){
            row.push(useCase[i,rowPosition]);
        }
        acum += searchRepeated(row, size);
        rowPosition += 1;
    }
    return acum;
}

function searchRepeated(array, size){
    for(let i=0; i<size; i++){
        for(let j=i+1; j<=size; j++){
            if(array[i] === array[j]){
                return 1;
            }
        }
    }
    return 0;
}

File: Vestigio/code.py
def getDataInt(message):
    data = int(input(message))
    return data

def searchRepeated(array, size):
    for i in range(size):
        for j in range(i + 1,size):
            if(array[i] == array[j]):
                return 1
    return 0

def getUseCase():
    useCase = []
    size = getDataInt('Enter the size of the matriz: ')
    for i in range(size):
        row = []
        for j in range(size):
            row.append(getDataInt('Enter the number in position ' + str(i) + ',' + str(j) + ': '))
        useCase.append(row)
    return useCase, size

def getTrace(useCase, size):
    sum = 0
    for i in range(size):
        sum += useCase[i][i]
    return sum

def validColum(useCase, size):
    acum = 0
    for i in range(size):
        acum += searchRepeated(useCase[i], size)
    return acum

def validRow(useCase, size):
    acum = 0
    rowPosition = 0
    while(rowPosition < size):
        row = []
        for i in range(size):
            row.append(useCase[i][rowPosition])
        acum += searchRepeated(row, size)
        rowPosition += 1
    return acum

count = getDataInt('Enter the number of use cases: ')
for i in range(count):
    useCase, size = getUseCase()
    trace = getTrace(useCase, size)
    colum = validColum(useCase, size)
    row = validRow(useCase, size)
    print('Use case Nº ' + str(i))
    print('Trace: ' + str(trace))
    print('Colum: ' + str(colum))
    print('Row: ' + str(row))